urlcaef331f4e_dreamable_cra_tgz-pJ8-coho_lp_project_LPProjectJ8/src/coho/lp/project/DoubleIntervalLPProject.java

```diff
@@ -0,0 +1,308 @@
+package coho.lp.project;
+
+import java.util.*;
+import coho.common.matrix.*;
+import coho.common.number.*;
+import coho.lp.*;
+import coho.lp.solver.*;
+import coho.geom.twodim.*;
+import coho.debug.*;
+
+public class DoubleIntervalLPProject extends LPProject {
+
+	DoubleIntervalLPProject(LP _lp, Matrix _x, Matrix _y) {
+		super(_lp, _x, _y, CohoSolverFactory.Solver.DOUBLEINTERVAL);
+	}
+	
+	private DoubleInterval nextNorm(DoubleIntervalCohoMatrix A_trans_base, Matrix x, Matrix y,double currNorm){
+		try{
+			DoubleIntervalMatrix pi = A_trans_base.getSolution(x);
+			DoubleIntervalMatrix eta = A_trans_base.getSolution(y); 
+			
+			DoubleInterval newNorm = DoubleInterval.create(Double.MAX_VALUE);
+			boolean nextQuad = true;
+			double minNextNorm = currNorm+planeEps*Math.max(1,currNorm);//sin(planeEps)*sqrt{1+currNorm^2}
+			for(int i=0;i<pi.length();i++){
+				DoubleInterval pi_i = pi.V(i);
+				DoubleInterval eta_i = eta.V(i);
+				if(eta_i.less(0.0)){
+					DoubleInterval Norm_i = pi_i.div(eta_i.negate());
+					double norm_i = Math.max(0,Norm_i.x().doubleValue());//make sure it's non-negative. For the special case (1,0).
+					//DOC What if Norm_i has greater interval? We use x() value for alpha.
+					if(norm_i >= minNextNorm && norm_i < newNorm.x().doubleValue()){
+						nextQuad = false;
+						double e = Norm_i.hi().doubleValue()-norm_i;
+						newNorm = DoubleInterval.create(norm_i-e,norm_i+e);//ususally it's Norm_i.
+					}
+				}
+			}
+			if(nextQuad||newNorm.x().doubleValue()>=1/planeEps)//if the angle is small then eps, turn to next quad.
+				return null;
+			return newNorm;//else
+		}catch(SingularMatrixException e){
+			throw new LPError("LPproject.nextNorm(): This should never happen when solve the optimal point.");
+		}
+	}	
+	public static final double planeEps = coho.geom.twodim.GeomObj2.eps;//smallest angle between two plane
+	public static final double minLength = 1e-6;//to test next quad. should be relative?
+	public static final double maxInc = Math.tan(Math.PI/2-planeEps);//the max value of alpha, otherwise, it should jump to next quad.
+	protected ArrayList<Line> findPlanes(Matrix A,Matrix b,Matrix x_axis,Matrix y_axis){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 25. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.